src/components/CompletedSubReqCourse.vue

```diff
@@ -1 +1,184 @@
-<!-- Component for a completed SubReqCourse-->
+<template>
+  <div class="completedsubreqcourse">
+    <div
+      v-for="courseObject in courseObjects"
+      :key="courseObject.uniqueID"
+      class="completed-reqCourses-course-wrapper"
+    >
+      <div id="completedSeparator" class="separator"></div>
+      <div class="completed-reqCourses-course-heading-wrapper">
+        <div class="completed-reqCourses-course-heading-course">
+          <span class="completed-reqCourses-course-heading-check"
+            ><img src="@/assets/images/checkmark-green.svg"
+          /></span>
+          {{ courseLabel }}
+        </div>
+        <div class="completed-reqCourses-course-heading-reset-button reqCourse-button">
+          {{ resetText }}
+        </div>
+      </div>
+      <div class="completed-reqCourses-course-object-wrapper">
+        <course
+          v-bind="courseObject"
+          :courseObj="courseObject"
+          :id="courseObject.subject + courseObject.number"
+          :uniqueID="courseObject.uniqueID"
+          :compact="true"
+          :active="false"
+          :isCompletedReqCourse="true"
+          :isReqCourse="true"
+          class="completed-reqCourses-course-object"
+        />
+        <div class="completed-reqCourses-course-object-semester">
+          {{ semesterLabel }}
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue, { PropType } from 'vue';
+import firebase from 'firebase/app';
+import Course from '@/components/Course.vue';
+import { DisplayableRequirementFulfillment, CourseTaken } from '@/requirements/types';
+import {
+  AppCourse,
+  FirestoreSemesterCourse,
+  AppSemester,
+  FirestoreSemesterType,
+} from '@/user-data';
+
+Vue.component('course', Course);
+
+type CrseInfo = {
+  roster: string;
+  crseIds: number[];
+};
+
+type CompletedSubReqCourseSlot = {
+  isCompleted: true;
+  courses: readonly CourseTaken[];
+};
+
+type IncompleteSubReqCourseSlot = {
+  isCompleted: false;
+  courses: CrseInfo[];
+};
```

Comment:
Can we standardize the types in some global file instead? Redefining them is kinda DRY

Reply:
Don't we already have @/requirements/types?

Reply:
Related comment: make arrays and field readonly if possible

Reply:
Good idea! I put these types in @/requirements/types. All fields are readonly except for `crseIds` since there seems to be direct manipulation of the `crseIds` field in `getAllCrseInfoFromSemester()`.